python/tvm/relax/transform/legalize_ops/ccl.py

```diff
@@ -46,6 +46,30 @@ def _allreduce(_bb: BlockBuilder, call: Call) -> Expr:
     )
 
 
+@register_legalize("relax.ccl.allgather")
+def _allgather(_bb: BlockBuilder, call: Call) -> Expr:
+    output_shape = []
+    assert isinstance(
+        call.args[0].struct_info, TensorStructInfo
```

Comment:
Since this expression occurs many times throughout the function, can we define `arg_sinfo = call.args[0].struct_info`?